src/RestQuery.js

```diff
@@ -700,11 +711,27 @@ RestQuery.prototype.handleIncludeAll = function() {
       this.include = [...new Set([...this.include, ...includeFields])];
       // if this.keys not set, then all keys are already included
       if (this.keys) {
-        this.keys = [...new Set([...this.keys, ...keyFields])];
+        this.keys = [...new Set([...this.keys, ...keyFields])]
+          // Ignore excluded keys
+          .filter(k => !this.excludeKeys || this.excludeKeys.indexOf(k) < 0);
       }
     });
 };
 
+// Upadates property `this.keys` to contain all keys but the ones unselected.
+RestQuery.prototype.handleExcludeKeys = function() {
+  if (!this.excludeKeys) {
+    return;
+  }
```

Comment:
Can you check for this.keys then load schema if needed?

Reply:
What do you mean?

Reply:
This function currently always loads the schema. If this.keys exist you don’t have to load the schema.